sagemaker/pipeline/pipeline_parameters.py

```diff
@@ -0,0 +1,764 @@
+        type=str,
+        required=True,
+        help="GraphStorm GConstruct/dist_part/train/inference CPU ECR image URL. Required",
+    )
+    optional_args.add_argument(
+        "--graphstorm-pytorch-gpu-image-url",
+        type=str,
+        help="GraphStorm GConstruct/dist_part/train/inference GPU ECR image URL.",
+    )
+    optional_args.add_argument(
+        "--gsprocessing-pyspark-image-url",
+        type=str,
+        help="GSProcessing SageMaker PySpark ECR image URL. Required if running GSProcessing",
+    )
+
+    # Instance Configuration
+    required_args.add_argument(
+        "--instance-count",
+        "--num-parts",
+        type=int,
+        help="Number of worker instances/partitions for partition, training, inference.",
+        required=True,
+    )
+    optional_args.add_argument(
+        "--cpu-instance-type",
+        type=str,
+        help=(
+            "CPU instance type. "
+            "Always used in DistPart step and if '--train-on-cpu' is provided, "
+            "in Train and Inference steps."
+        ),
+        default="ml.m5.4xlarge",
+    )
+    optional_args.add_argument(
+        "--gpu-instance-type",
+        type=str,
+        help=(
+            "GPU instance type. Used by default in in Train and Inference steps, "
+            "unless '--train-on-cpu' is set."
+        ),
+        default="ml.g5.4xlarge",
+    )
+    optional_args.add_argument(
+        "--train-on-cpu",
+        action="store_true",
+        help="Run training and inference on CPU instances instead of GPU.",
+    )
+    optional_args.add_argument(
+        "--graph-construction-instance-type",
+        type=str,
+        default=None,
+        help=(
+            "Instance type for graph construction. "
+            "Used in GSProcessing, GConstruct, GraphBoltConversion steps. "
+            "Default: same as CPU instance type"
+        ),
+    )
+    optional_args.add_argument(
+        "--gsprocessing-instance-count",
+        type=int,
+        default=None,
+        help="Number of GSProcessing instances. Default is equal to number of training instances.",
+    )
+    optional_args.add_argument(
+        "--volume-size-gb",
+        type=int,
+        help="Additional volume size for instances in GB.",
+        default=100,
+    )
+
+    # Pipeline/Task Configuration
+    required_args.add_argument(
+        "--graph-name", type=str, required=True, help="Name of the graph. Required."
+    )
+    required_args.add_argument(
+        "--input-data-s3",
+        type=str,
+        required=True,
+        help="S3 path to the input graph data. Required.",
+    )
+    required_args.add_argument(
+        "--output-prefix-s3",
+        type=str,
+        required=True,
+        help="S3 prefix for the output data. Required.",
+    )
+    optional_args.add_argument(
+        "--pipeline-name",
+        type=str,
+        help="Name for the pipeline. Needs to be unique per account and region.",
```

Comment:
What if it is not provided?

Reply:
We use the function `_get_pipeline_name(args: PipelineArgs) -> str` which covers both cases.